File: data/src/main/java/org/cryptomator/data/cloud/pcloud/RootPCloudFolder.java
package org.cryptomator.data.cloud.pcloud;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.PCloud;

class RootPCloudFolder extends PCloudFolder {

	private final PCloud cloud;

	public RootPCloudFolder(PCloud cloud) {
		super(null, "", "");
		this.cloud = cloud;
	}

	@Override
	public PCloud getCloud() {
		return cloud;
	}

	@Override
	public PCloudFolder withCloud(Cloud cloud) {
		return new RootPCloudFolder((PCloud) cloud);
	}
}

File: data/src/apiKey/java/org/cryptomator/data/cloud/pcloud/RootPCloudFolder.kt
package org.cryptomator.data.cloud.pcloud

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.PCloud

internal class RootPCloudFolder(override val cloud: PCloud) : PCloudFolder(null, "", "") {

	override fun withCloud(cloud: Cloud?): PCloudFolder {
		return RootPCloudFolder(cloud as PCloud)
	}
}
